convert2rhel/unit_tests/checks_test.py

```diff
@@ -438,14 +439,34 @@ def test_ensure_compatibility_of_kmods(
         assert shouldnt_be_in_logs not in caplog.records[-1].message
 
 
-def test_validate_package_manager_transaction(monkeypatch, caplog):
+@centos8
+def test_ensure_compatibility_of_kmods_check_env(
+    monkeypatch,
+    pretend_os,
+    caplog,
+):
+
+    monkeypatch.setattr(os, "environ", {"CONVERT2RHEL_UNSUPPORTED_UNCHECKED_KMODS": "1"})
+    monkeypatch.setattr(checks, "get_loaded_kmods", mock.Mock(return_value=HOST_MODULES_STUB_BAD))
+    run_subprocess_mock = mock.Mock(
+        side_effect=run_subprocess_side_effect(
+            (("uname",), ("5.8.0-7642-generic\n", 0)),
+            (("repoquery", "-f"), (REPOQUERY_F_STUB_GOOD, 0)),
+            (("repoquery", "-l"), (REPOQUERY_L_STUB_GOOD, 0)),
+        )
+    )
     monkeypatch.setattr(
-        checks.pkgmanager,
-        "create_transaction_handler",
-        value=mock.Mock(),
+        checks,
+        "run_subprocess",
+        value=run_subprocess_mock,
     )
 
-    checks.validate_package_manager_transaction()
+    checks.ensure_compatibility_of_kmods()
+    should_be_in_logs = (
+        ".*Detected 'CONVERT2RHEL_UNSUPPORTED_UNCHECKED_KMODS' environment variable."
+        " We will continue the conversion, with the following kernel modules that are not supported:.*"
+    )
+    assert re.match(should_be_in_logs in caplog.records[-1].message, re.MULTILINE, re.DOTALL)
```

Comment:
Your unit tests are failing right now because of this regex, there are a couple of things that need to be changed. 

First, the `re.match` accepts an `pattern`, one `string`, and some `flags`. The way you wrote that was partially correct, except from a few things that were missing, below I have a suggestion that fixes this, but let me explain what changed as well.

```suggestion
    assert re.match(pattern=should_be_in_logs, string=caplog.records[-1].message, flags=re.MULTILINE | re.DOTALL)
```

Instead of having an `pattern` and `string`, you were using only “one” parameter that acted as the `pattern` (`should_be_in_logs in caplog.records[-1].message`), probably this was a leftover after refactoring the unit tests in a previous commit, so no worries. 

Lastly, you are passing the flags to `re.match` as multiple arguments `re.match(..., re.MULTILINE, re.DOTALL)`, kind of something like `re.match(..., flag1=re.MULTILINE, flag=2re.DOTALL)` (just an example, `flag1` and `flag2` doesn't exist), while the correct way of passing more than one flag to `re.match` is by using the `|` operator, like this `re.match(..., flags=re.MULTILINE | re.DOTALL)`. 

Hope you can apply the changes easily and hopefully all unit tests will pass now!